### DISCUSSION TITLE
Why tun2socks in `fd` device mode acceidentally closes the socket

### DISCUSSION THREAD

**Discussion by Alkenso** [xjasonlyu/tun2socks#247] [General]
For some reason, I need to feed IP-level packets to tun2socks tool via UNIX socket
Initially packets going well, but after short period of time tun2socks closes this socket and don't want to receive/send any data with it.

Why it may happen?

**Comment by xjasonlyu** [comment]
No idea on my side, you probably need to provider more info about this.
